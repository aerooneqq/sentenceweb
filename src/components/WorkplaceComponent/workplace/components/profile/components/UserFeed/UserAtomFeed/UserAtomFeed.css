.userAtomFeedCont { 
    width: 100%; 
    height: auto;
    background: transparent;

    border-style: none;

    margin-top: 30px;

    display: flex;
    flex-direction: row;
    justify-content: center; 
    align-items: center;
}

.userAtomFeedPhoto {  
    width: 51px;
    height: 51px;

    padding: 0.23px;

    box-sizing: content-box;

    border-style: solid;
    border-width: 1px;
    border-radius: 50px;
    border-color: #F97646;

    margin-left: 10px;
}

.userFeedTextCont { 
    width: 350px;
    margin-left: 10px;
}

.userAtomFeedCont:nth-child(1) { 
    margin-top: 10px;
}

.userAtomFeedCont:nth-last-child(1) { 
    margin-bottom: 10px;
}

.userAtomFeedData { 
    width: 100%;
    height: auto;

    display: flex;
    flex-direction: row;
}

.userAtomFeedAuthorName { 
    color: gray;
    font-size: 14px;
}

.userAtomFeedText { 
    margin-top: 5px;

    font-family: Arial; 
    color: #404040;
    font-size: 16px;
}

.userAtomFeedDate { 
    margin-left: 10px;

    font-family: Arial; 
    color: gray;
    font-size: 14px;
}