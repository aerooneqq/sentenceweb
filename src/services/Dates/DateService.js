/**
 * The Dates Service used to get the parts of the date which is encoded in the following format:
 * YYYY:MM:DDTHH:MM:SS+NN:NN 
 */
export default class DateService { 

    constructor(date) { 
        this.date = date;
    }

    getYear() { 
        return Number(this.date.substr(0, 4));
    }

    getMonth() {
        let month = this.date.substr(5, 2);
        while (month.charAt(0) === "0") {
            month = month.substr(1)
        }

        return Number(month);
    }

    getDay() { 
        let day = this.date.substr(8, 2)
        while (day.charAt(0) === "0"){ 
            day = day.substr(1);
        }

        return Number(day);
    }

    getTime() { 
        return this.date.substr(11, 8);
    }
}