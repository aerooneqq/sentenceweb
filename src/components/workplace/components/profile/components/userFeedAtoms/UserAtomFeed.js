import React, {Component} from "react"
import "./styles/UserAtomFeed.css"

export default class UserAtomFeed extends Component{ 
    constructor(props){ 
        super(props)
    }

    render(){ 
        return( 
            <div className="userAtomFeedCont">
                <div className="userAtomFeedPhoto">

                </div>

                <div className="userFeedTextCont">
                    <div className="userAtomFeedText">
                        D Descr iptionesDes criptDes cription ion criDescrip t ionDe script ionpt DecriDe scrip sc ripcriDescrip tionion
                    </div>
                    <div className="userAtomFeedDate">
                        30.05.2000
                    </div>
                </div>
            </div>
        )
    }
}