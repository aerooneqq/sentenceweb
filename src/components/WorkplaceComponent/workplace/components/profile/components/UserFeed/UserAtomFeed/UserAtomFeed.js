import React, {Component} from "react"

//Styles
import "./UserAtomFeed.css"
import DateService from "../../../../../../../../services/dates/DateService";

export default class UserAtomFeed extends Component{ 
    constructor(props){ 
        super(props)

        let dateService = new DateService(props.date);

        let day = dateService.getDay();
        let month = dateService.getMonth();
        let year = dateService.getYear();

        this.state = { 
            message: props.message,
            date: day + "." + month + "." + year
        };
    }

    render(){ 
        return( 
            <div className="userAtomFeedCont">
                <div className="userAtomFeedPhoto">

                </div>

                <div className="userFeedTextCont">
                    <div className="userAtomFeedText">
                        {this.state.message}
                    </div>
                    <div className="userAtomFeedDate">
                        {this.state.date}
                    </div>
                </div>
            </div>
        )
    }
}