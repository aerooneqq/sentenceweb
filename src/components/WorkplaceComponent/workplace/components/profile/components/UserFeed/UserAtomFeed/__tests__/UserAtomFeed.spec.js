import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import UserAtomFeed from "../UserAtomFeed";
import DateService from "../../../../../../../../../services/Dates/DateService";

describe("UserAtomFeed", () => { 
    it("should render correctly", () => { 
        const props = { 
            message: "test message"
        }

        expect(shallowToJson(shallow(<UserAtomFeed />))).toMatchSnapshot();
    });

    it("should not crash when props are null", () => { 
        const props = { 
            message: null,
            date: null
        }

        shallow(<UserAtomFeed {...props}/>);
    });

    it("should not crash when props are undefined", () => { 
        const props = { 
            message: undefined,
            date: undefined,
        }

        shallow(<UserAtomFeed {...props}/>)
    });

    it("should have proper state", () => { 
        const props = { 
            message: "test message"
        }

        const properState = { 
            message: props.message,
        }

        const userAtomFeed = shallow(<UserAtomFeed {...props}/>)

        for (let key in properState) { 
            expect(properState[key]).toBe(userAtomFeed.state(key));
        }
    });
});