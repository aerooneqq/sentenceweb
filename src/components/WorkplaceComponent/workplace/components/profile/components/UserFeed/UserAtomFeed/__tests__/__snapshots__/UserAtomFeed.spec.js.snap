// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAtomFeed should render correctly 1`] = `
<div
  className="userAtomFeedCont"
>
  <div
    className="userAtomFeedPhoto"
  >
    <UserFeedPhoto />
  </div>
  <div
    className="userFeedTextCont"
  >
    <div
      className="userAtomFeedData"
    >
      <div
        className="userAtomFeedAuthorName"
      />
      <div
        className="userAtomFeedDate"
      >
        77.93.1570
      </div>
    </div>
    <div
      className="userAtomFeedText"
    />
  </div>
</div>
`;
